volumes.c

```diff
@@ -92,19 +124,108 @@ void pv_volumes_empty(struct pv_state *s)
 	pv_log(INFO, "removed %d volumes", num_vol);
 }
 
-struct pv_volume* pv_volume_add(struct pv_state *s, char *name)
+struct pv_volume* pv_volume_add(struct pv_state *s, char *name, char *disk)
 {
 	struct pv_volume *v = calloc(1, sizeof(struct pv_volume));
+	struct pv_disk *d, *tmp;
+	struct dl_list *disks = NULL;
 
 	if (v) {
 		v->name = strdup(name);
 		dl_list_init(&v->list);
 		dl_list_add_tail(&s->volumes, &v->list);
+		disks = &s->disks;
+
+		if (disk) {
+			dl_list_for_each_safe(d, tmp, disks, struct pv_disk, list) {
+				if (!strcmp(d->name, disk)) {
+					v->disk = d;
+					break;
+				}
+			}
+		}
+		else {
+			dl_list_for_each_safe(d, tmp, disks, struct pv_disk, list) {
+				if (d->def) {
+					v->disk = d;
+					break;
+				}
+			}
+		}
 	}
 
 	return v;
 }
 
+struct pv_disk* pv_disk_add(struct pv_state *s)
+{
+	struct pv_disk *d = calloc(1, sizeof(struct pv_disk));
+
+	if (d) {
+		dl_list_init(&d->list);
+		dl_list_add_tail(&s->disks, &d->list);
+	}
+
+	return d;
+}
+
+static int pv_volumes_unmount_handler(struct pantavisor *pv, struct pv_volume *v)
```

Comment:
when is unmount handler called in the lifecycle? only pre shut down?
Needs implementation ...

Reply:
`pv_volumes_mount_handler` is the only function needed now with action type as mount/umount. Updated it now.